pandas/core/window/rolling.py

```diff
@@ -145,6 +145,20 @@ def _validate_get_window_bounds_signature(window: BaseIndexer) -> None:
                 f"get_window_bounds"
             )
 
+    @staticmethod
+    def _validate_baseindexer_support(window, func_name: Optional[str]) -> None:
```

Comment:
either make this a method or just a module level free function ; i find static methods are
non obvious to grok

Reply:
Done. Created a common function instead of a static method